axelrod/strategies/axelrod_second.py

```diff
@@ -199,3 +202,194 @@ def strategy(self, opponent: Player) -> Action:
+    """
+
+    name = 'Tranquilizer'
+    classifier = {
+        'memory_depth': float('inf'),
+        'stochastic': True,
+        'makes_use_of': {"game"},
+        'long_run_time': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def __init__(self):
+        super().__init__()
+        self.num_turns_after_good_defection = 0 # equal to FD variable in Fortran code
+        self.opponent_consecutive_defections = 0
+        self.one_turn_after_good_defection_ratio= 5
+        self.two_turns_after_good_defection_ratio= 0
+        self.one_turn_after_good_defection_ratio_count = 1
+        self.two_turns_after_good_defection_ratio_count = 1
+        self.current_score = 0
+        self.dict = {C: 0, D: 1}
+
+
+    def update_cooperative_state(self, opponent):  
+        
+        """
+        Calculates the ratio values for the one_turn_after_good_defection_ratio,
+        two_turns_after_good_defection_ratio and the probability values, 
+        as well as sets the value of num_turns_after_good_defection.
+        """
+        if opponent.history[-1] == D: 
+            self.opponent_consecutive_defections += 1
+        else:
+            self.opponent_consecutive_defections = 0
+
+        if self.num_turns_after_good_defection == 2:
+            self.num_turns_after_good_defection = 0
+            self.two_turns_after_good_defection_ratio = (
+                ((self.two_turns_after_good_defection_ratio 
+                * self.two_turns_after_good_defection_ratio_count) 
+                + (3 - (3 * self.dict[opponent.history[-1]])) 
+                + (2 * self.dict[self.history[-1]]) 
+                - ((self.dict[opponent.history[-1]] 
+                * self.dict[self.history[-1]]))) 
+                / (self.two_turns_after_good_defection_ratio_count + 1)
+                )
+            self.two_turns_after_good_defection_ratio_count += 1
+        elif self.num_turns_after_good_defection == 1:
+            self.num_turns_after_good_defection = 2
+            self.one_turn_after_good_defection_ratio = (
+                ((self.one_turn_after_good_defection_ratio 
+                * self.one_turn_after_good_defection_ratio_count)
+                + (3 - (3 * self.dict[opponent.history[-1]])) 
+                + (2 * self.dict[self.history[-1]]) 
+                - (self.dict[opponent.history[-1]] 
+                * self.dict[self.history[-1]])) 
+                / (self.one_turn_after_good_defection_ratio_count + 1)
+                )
+            self.one_turn_after_good_defection_ratio_count += 1
+   
+    def strategy(self, opponent: Player) -> Action:
+
+        if len(self.history) == 0:
+            return C
+
+        self.current_score = compute_final_score(zip(self.history, opponent.history))
+       
+        self.update_cooperative_state(opponent)
+        if  self.num_turns_after_good_defection in [1, 2]:
+            return C        
+        
```

Comment:
Let's move `self.current_score = compute_final_score(zip(self.history, opponent.history))` to here. IE move it to after the code block that returns `C` when we're in a cooperative state.